ci-operator/step-registry/ipi/conf/gcp/xpn/ipi-conf-gcp-xpn-commands.sh

```diff
@@ -4,13 +4,104 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
-CONFIG="${SHARED_DIR}/install-config.yaml"
-PATCH=/tmp/install-config-xpn.yaml.patch
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
 
+fi
+
+XPN_SERVICE_ACCOUNT=`gcloud config get-value account`
 HOST_PROJECT="openshift-dev-installer"
-HOST_PROJECT_CONTROL_SUBNET="installer-shared-vpc-subnet-1"
-HOST_PROJECT_COMPUTE_SUBNET="installer-shared-vpc-subnet-2"
-HOST_PROJECT_NETWORK="installer-shared-vpc"
+# set the project in order to create the resources
+gcloud config set project "${HOST_PROJECT}"
+
+echo "$(date -u --rfc-3339=seconds) - Configuring GCP Subnets for Host Project..."
+
+REGION="${LEASED_RESOURCE}"
+INSTANCE_PREFIX="${NAMESPACE}-${JOB_NAME_HASH}"
+HOST_PROJECT_CONTROL_SUBNET="${INSTANCE_PREFIX}-subnet-1"
+HOST_PROJECT_COMPUTE_SUBNET="${INSTANCE_PREFIX}-subnet-2"
+HOST_PROJECT_NETWORK="${INSTANCE_PREFIX}-vpc"
+CONTROL_SUBNET_CIDR="10.0.64.0/19"
+COMPUTE_SUBNET_CIDR="10.0.128.0/19"
```

Comment:
This is fine for the PoC, but I think having these as static values will prevent multiple jobs from running simultaneously. The subnets will need to have non-overlapping CIDRs, so we will need to manage these and make the values populated in the install config dynamic.